core/src/main/scala/kafka/server/AutoTopicCreationManager.scala

```diff
@@ -166,7 +179,34 @@ class DefaultAutoTopicCreationManager(
         debug(s"Auto topic creation completed for ${creatableTopics.keys}.")
         clearInflightRequests(creatableTopics)
       }
-    })
+    }
+
+    val channelManager = this.channelManager.getOrElse {
+      throw new IllegalStateException("Channel manager must be defined in order to send CreateTopic requests.")
+    }
+
+    metadataRequestContext match {
+      case Some(context) =>
+        val requestVersion =
+          channelManager.controllerApiVersions() match {
+            case None =>
+              ApiKeys.CREATE_TOPICS.latestVersion()
+            case Some(nodeApiVersions) =>
+              nodeApiVersions.latestUsableVersion(ApiKeys.CREATE_TOPICS)
+          }
+
+        // Borrow client information such as client id and correlation id from the original request,
+        // in order to correlate the create request with the original metadata request.
+        val requestHeader = new RequestHeader(ApiKeys.CREATE_TOPICS,
+          requestVersion,
+          context.clientId,
+          context.correlationId)
+        val envelopeRequest = ForwardingManager.buildEnvelopeRequest(context,
+          createTopicsRequest.build().serializeWithHeader(requestHeader))
+        channelManager.sendRequest(envelopeRequest, requestCompletionHandler)
```

Comment:
nit: this call is shared by both branches, maybe we could simplify a little bit. something like this:
```scala
val request = metadataRequestContext.map { context =>
  ...
}.getOrElse(createTopicRequest)

channelManager.sendRequest(request, requestCompletionHandler)
```